### DISCUSSION TITLE
Add multline separation

### DISCUSSION THREAD

**Discussion by RegulumDreik** [sevdokimov/log-viewer#11] [Ideas]
Hello! I am very interested in this application. But I got a problem when using it with multi-line logs. It is not possible to adequately separate messages with more than one line in them. In filebeats, as far as I know, it is possible to write a pattern for dividing the log into messages. For further analysis. I would like to see similar functionality here.

**Comment by sevdokimov** [comment]
Could you attach an example of the multiline log?

**Reply by RegulumDreik** [reply]
That is, the log contains time, technical information, and an XML message.
I parsed this messages by regexp, but regexp not fill timestamp.

```
[18.11.19 18:26:22.160]	00001720	Receive: ParsedData 
<DateEvent Year="2019" Month="11" Day="28" Hour="18" Minute="26" Second="22" Milliseconds="160" /> 
<Story	ID="1"> 
	<Actions> 
			<JUMP	Name="Kirill"	ActionTime="1913" /> 
			<RUN	Name="Petya"	ActionTime="2814" /> 
			<RUN	Name="Vasya"	ActionTime="2001" /> 
			<JUMP	Name="Igor"	     ActionTime="32350" /> 
			<JUMP	Name="Denis"	ActionTime="18250" /> 
			<STOP	Name="Kirill"	ActionTime="1913" /> 
	</Actions> 
</Story> 
<StoryEnd story ID="1" /> 
```

```
[18.11.19 18:26:22.160]	 <Start point="26" story="1" /> 
[18.11.19 18:26:22.162]	 <Req point="26" story="1" seq="1540" Time="2000" > 
    <Log seq="1"> 
NUMBER 5 
 1 Kirill 
 2 Petya 
 4 Vasya 
 5 Igor 
 6 Egor 

LEVEL1 

JUMP Kirill 
RUN Petya 
STOP Vasya 
JUMP Igor 
JUMP Egor 
[18.11.19 18:27:02.176]	<End point="26" story="1" /> 
```

**Comment by sevdokimov** [comment]
Yes, it's possible to separate messages correctly. You have to specify the format of the log manually in `log-viewer-0.1.5/config.conf`. There is `logs = [ ...  ]` section that lists available logs and its format. Add the following lines here

```
    {
        path: "/my-app/logs/filename.log"
        format: {
            type: Log4jLogFormat,
            pattern: "[%d{dd.MM.yy HH:mm:ss.SSS}]\t%m%n"
        }
    }

```

Don't forget to replace "/my-app/logs/filename.log" with the path to your log location.

The result will look like 
```
logs = [
  {
    path: "**/*.log"  // All files with ".log" extension are visible by default
  }
  {
    path: "**/*.log.*"  // All files like "*.log.1" are visible by default
  }
  {
    path: "/my-app/logs/filename.log"
    format: {
      type: Log4jLogFormat,
      pattern: "[%d{dd.MM.yy HH:mm:ss.SSS}]\t%m%n"
    }
  }
]

```
